src/printer.js

```diff
@@ -3702,14 +3700,20 @@ function printMemberChain(path, options, print) {
     printIndentedGroup(groups.slice(shouldMerge ? 2 : 1))
   ]);
 
-  // If there's a comment, we don't want to print in one line.
-  if (hasComment) {
-    return group(expanded);
-  }
+  const callExpressionCount = printedNodes.filter(
+    tuple => tuple.node.type === "CallExpression"
+  ).length;
 
-  // If any group but the last one has a hard line, we want to force expand
-  // it. If the last group is a function it's okay to inline if it fits.
-  if (printedGroups.slice(0, -1).some(willBreak)) {
+  // We don't want to print in one line if there's:
+  //  * A comment.
+  //  * 3 or more chained calls.
+  //  * Any group but the last one has a hard line.
+  // If the last group is a function it's okay to inline if it fits.
+  if (
+    hasComment ||
+    callExpressionCount >= 3 ||
```

Comment:
I really need this thing to be configurable, always print in one line in certain files.

Reply:
Would you be able to share some code where this produces undesirable output?

Reply:
@azz particularly in db migration files: https://github.com/kriasoft/nodejs-api-starter/blob/master/migrations/201612010000_initial.js

Reply:
FWIW, I'd personally find those more readable if multiline.

Reply:
Seeing the length of you lines, I'm pretty sure that Prettier <1.6 already place the calls on separate lines.

[Here's](https://prettier.io/playground/#%7B%22content%22%3A%22db.schema.createTable('story_points'%2C%20table%20%3D%3E%20%7B%5Cn%20%20%20%20table.uuid('story_id').references('id').inTable('stories').onDelete('CASCADE').onUpdate('CASCADE')%3B%20%2F%2F%20prettier-ignore%5Cn%20%20%20%20table.uuid('user_id').notNullable().references('id').inTable('users').onDelete('CASCADE').onUpdate('CASCADE')%3B%20%2F%2F%20prettier-ignore%5Cn%20%20%20%20table.primary(%5B'story_id'%2C%20'user_id'%5D)%3B%5Cn%20%20%7D)%3B%5Cn%5Cndb.schema.createTable('story_points'%2C%20table%20%3D%3E%20%7B%5Cn%20%20%20%20table.uuid('story_id').references('id').inTable('stories').onDelete('CASCADE').onUpdate('CASCADE')%3B%5Cn%20%20%5Cn%20%20%20%20table.uuid('user_id').notNullable().references('id').inTable('users').onDelete('CASCADE').onUpdate('CASCADE')%3B%5Cn%20%20%5Cn%20%20%20%20table.primary(%5B'story_id'%2C%20'user_id'%5D)%3B%5Cn%20%20%7D)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D) your example with and without prettier-ignore.

The second solution looks much more readable and maintainable.

Reply:
@SimenB @azz image a single DB table with just a few columns but not fitting onto a single screen forcing you to scroll the page in the editor :(

Reply:
You could just `prettier-ignore` the whole `exports.up` block. You're not getting a huge benefit from using prettier in a database migration file.

Reply:
@ValentinH this [`migrations/.eslintrc`](https://github.com/kriasoft/nodejs-api-starter/blob/169fb68a12772246b0ae245cb745ff532e024296/migrations/.eslintrc#L4-L11) file worked well for me with Prettier <1.6

Reply:
Indeed, you are using a large line width so that probably helps :)